src/core/IPosisProcess.d.ts

```diff
@@ -3,6 +3,8 @@ interface IPosisProcess {
     imageName: string; // image name (maps to constructor)
     id: PosisPID; // ID
     parentId: PosisPID; // Parent ID
-    log: IPosisLogger; // Logger 
+    log: IPosisLogger; // Logger
+    // For querying extension interfaces (instead of tying ourselves to "levels")
+    queryPosisInterface<T extends keyof PosisInterfaces>(interfaceId: T): PosisInterfaces[T] | undefined;
```

Comment:
This is exactly the same as it is with the other properties of `IPosisProcess` (except `run`), I don't see any reason why it would be a bad idea to do the same for `queryPosisInterface`.